dm/pkg/checker/table_structure.go

```diff
@@ -267,114 +336,198 @@ func (c *TablesChecker) checkTableOption(opt *ast.TableOption) *incompatibilityO
 // * check whether they have same column list
 // * check whether they have auto_increment key.
 type ShardingTablesChecker struct {
-	name string
-
+	targetTableID                string
 	dbs                          map[string]*sql.DB
-	tables                       map[string]map[string][]string // instance => {schema: [table1, table2, ...]}
+	tableMap                     map[string][]*filter.Table // instance => {[table1, table2, ...]}
 	mapping                      map[string]*column.Mapping
 	checkAutoIncrementPrimaryKey bool
 }
 
 // NewShardingTablesChecker returns a RealChecker.
-func NewShardingTablesChecker(name string, dbs map[string]*sql.DB, tables map[string]map[string][]string, mapping map[string]*column.Mapping, checkAutoIncrementPrimaryKey bool) RealChecker {
+func NewShardingTablesChecker(targetTableID string, dbs map[string]*sql.DB, tableMap map[string][]*filter.Table, mapping map[string]*column.Mapping, checkAutoIncrementPrimaryKey bool) RealChecker {
 	return &ShardingTablesChecker{
-		name:                         name,
+		targetTableID:                targetTableID,
 		dbs:                          dbs,
-		tables:                       tables,
+		tableMap:                     tableMap,
 		mapping:                      mapping,
 		checkAutoIncrementPrimaryKey: checkAutoIncrementPrimaryKey,
 	}
 }
 
 // Check implements RealChecker interface.
-func (c *ShardingTablesChecker) Check(ctx context.Context) *Result {
+func (c *ShardingTablesChecker) Check(ctx context.Context) (*Result, error) {
 	r := &Result{
 		Name:  c.Name(),
 		Desc:  "check consistency of sharding table structures",
 		State: StateSuccess,
-		Extra: fmt.Sprintf("sharding %s", c.name),
+		Extra: fmt.Sprintf("sharding %s,", c.targetTableID),
 	}
-
+	log.L().Logger.Info("start to check sharding tables")
 	var (
 		stmtNode      *ast.CreateTableStmt
 		firstTable    string
 		firstInstance string
+		errCh         = make(chan error, 1)
+		inChs         = make(map[string]chan *filter.Table, len(c.tableMap))
+		checkWg       sync.WaitGroup
 	)
 
-	for instance, schemas := range c.tables {
+	// get first table
+	for instance, tables := range c.tableMap {
```

Comment:
can we also do this in `checkFunc`, not sure if there will be too many shards(such as 200 source) want to check 🤔

Reply:
just execute once whatever source numbers.